File: programmers/js/level00/120853.js
// 컨트롤 제트
function solution(s) {
  const sArr = s.split(' ');
  
  while (sArr.includes('Z')) {
    sArr.splice(sArr.indexOf('Z') - 1, 2);
  }
         
  return sArr.reduce((acc, cur) => +acc + +cur, 0);
}

// 다른 풀이
// function solution(s) {
//   const stack = []

//   s.split(' ').forEach((target) => {
//     if(target === 'Z') stack.pop();
//     else stack.push(+target)
//   })

//   return stack.reduce((pre, cur) => pre + cur, 0);
// }

File: programmers/java/level00/120853.java
// 컨트롤 제트

import java.util.Arrays;

class Solution {
    public int solution(String s) {
        String[] sArr = s.split(" ");
        int result = 0;
        
        for (int i = 0; i < sArr.length; i++) {
            if (sArr[i].equals("Z")) {
                result -= Integer.parseInt(sArr[i - 1]);
            } else {
                result += Integer.parseInt(sArr[i]);
            }
        }
        
        return result;
    }
}